
<launch>
  <node pkg="open_quadtree_mapping" type="open_quadtree_mapping_tf_node" name="open_quadtree_mapping" clear_params="true" output="screen" >

    <param name="cam_width" value="1241" />
    <param name="cam_height" value="376" />
    <param name="cam_fx" value="620.5000000000001" />
    <param name="cam_cx" value="620.5" />
    <param name="cam_fy" value="620.5000000000001" />
    <param name="cam_cy" value="188.0" />
    <param name="ego_to_cam" value="0.27 0.0 1.65 0.5 -0.5 0.5 -0.5" />
    <!--downsample factor: 0 ~ 1-->
    <param name="downsample_factor" value="1"/>
    <param name="doBeliefPropagation" value="false" />
    <param name="useQuadtree" value="false" />
    <param name="doFusion" value="true" />
    <param name="doGlobalUpsampling" value="false" />
    <param name="semi2dense_ratio" value="1" />
    <param name="cost_downsampling" value="4" />
    <param name="min_inlier_ratio_good" value="0.49" />
    <param name="min_inlier_ratio_bad" value="0.4" />
    <param name="prev_variance_factor" value="1.0" />
    <param name="new_variance_factor" value="1.0" />
    <param name="variance_offset" value="0.0" />
    <param name="min_depth" value="1.0"/>
    <param name="max_depth" value="25.0" />
    <param name="inverse_depth" value="false" />
    <param name="new_keyframe_max_distance" value="0.5" />
    <param name="new_keyframe_max_angle" value="10.0" />
    <param name="new_reference_max_distance" value="0.03" />
    <param name="new_reference_max_angle" value="5.0" />
    <param name="p1" value="0.0006" /> <!-- original p1 = 0.003, p2 = 0.01-->
    <param name="p2" value="0.001" />

    <param name="new_reference_max_distance" value="0.5" />

    <param name="debug_to_folder" value="false" />

    <param name="use_tf_lookup" value="false" />
    <param name="target_frame" value="/ego_vehicle" />
    <param name="source_frame" value="/ego_vehicle/camera/rgb/left" />

    <!-- if using synchronize image and pose-->
    <remap from="~posestamped" to="/transform_center" />
    <!-- if using synchronize image and pose-->
    <remap from="~image" to="/carla/ego_vehicle/camera/rgb/left/image_raw" />

  </node>
</launch>
